app/package/controllers.py

```diff
@@ -330,70 +331,66 @@ def purge_data_package(publisher, package):
         return handle_error('GENERIC_ERROR', e.message, 500)
 
 
-@package_blueprint.route("/<publisher>/<package>/finalize",
-                         methods=["POST"])
+@package_blueprint.route("/upload", methods=["POST"])
 @requires_auth
-@is_allowed('Package::Create')
-def finalize_metadata(publisher, package):
+def finalize_metadata():
```

Comment:
I'd have renamed the method to something more meaningful. It is not finalizing metadata so much as completing the "publish process" so maybe:

```
finalize_publish
```